src/main/kotlin/com/openlattice/organizations/OrganizationExternalDatabaseController.kt

```diff
@@ -0,0 +1,197 @@
+        aclKeys.forEach { aclKey ->
+            ensureOwnerAccess(aclKey)
+            authorizations.deletePermissions(aclKey)
+            securableObjectTypes.deleteSecurableObjectType(aclKey)
+        }
+        edms.deleteOrganizationExternalDatabaseTables(organizationId, tableNameById)
+    }
+
+    @Timed
+    @DeleteMapping(path = [ID_PATH + TABLE_NAME_PATH + COLUMN_NAME_PATH + EXTERNAL_DATABASE_COLUMN])
+    override fun deleteExternalDatabaseColumn(
+            @PathVariable(OrganizationsApi.ID) organizationId: UUID,
+            @PathVariable(OrganizationsApi.TABLE_NAME) tableName: String,
+            @PathVariable(OrganizationsApi.COLUMN_NAME) columnName: String
+    ) {
+        val tableId = getExternalDatabaseObjectId(organizationId, tableName)
+        val columnId = getExternalDatabaseObjectId(tableId, columnName)
+        val aclKey = AclKey(organizationId, tableId, columnId)
+        ensureOwnerAccess(aclKey)
+        ensureObjectCanBeDeleted(tableId)
+        authorizations.deletePermissions(aclKey)
+        securableObjectTypes.deleteSecurableObjectType(aclKey)
+        edms.deleteOrganizationExternalDatabaseColumn(organizationId, tableName, columnName, columnId)
+    }
+
+    @Timed
+    @DeleteMapping(path = [ID_PATH + TABLE_NAME_PATH + EXTERNAL_DATABASE_COLUMN])
+    override fun deleteExternalDatabaseColumns(
+            @PathVariable(OrganizationsApi.ID) organizationId: UUID,
+            @PathVariable(OrganizationsApi.TABLE_NAME) tableName: String,
+            @RequestBody columnNames: Set<String>
+    ) {
+        val tableId = getExternalDatabaseObjectId(organizationId, tableName)
+        val columnNameById = columnNames.stream().collect<Map<UUID, String>, Any>(
+                Collectors.toMap({ columnName -> getExternalDatabaseObjectId(tableId, columnName) },
+                        { columnName -> columnName }))
+        val aclKeys = columnNameById.keys.stream().map { columnId -> AclKey(organizationId, tableId, columnId) }
+                .collect<Set<AclKey>, Any>(Collectors.toSet())
+        columnNameById.forEach { (id, name) -> ensureObjectCanBeDeleted(id) }
+        aclKeys.forEach { aclKey ->
+            ensureOwnerAccess(aclKey)
+            authorizations.deletePermissions(aclKey)
+            securableObjectTypes.deleteSecurableObjectType(aclKey)
+        }
+        edms.deleteOrganizationExternalDatabaseColumns(organizationId, tableName, columnNameById)
+    }
+
+    private fun ensureOwner(organizationId: UUID): AclKey {
+        val aclKey = AclKey(organizationId)
+        accessCheck(aclKey, EnumSet.of(Permission.OWNER))
+        return aclKey
+    }
+
+    private fun getExternalDatabaseObjectId(containingObjectId: UUID, name: String): UUID {
+        val fqn = FullQualifiedName(containingObjectId.toString(), name)
+        val id = aclKeysMap.get(fqn.fullQualifiedNameAsString)
+        checkState(id != null, "External database object with name $name does not exist")
```

Comment:
I feel like this should never happen but also Kotlin hates it without the nullcheck 🤔

Reply:
And even after checking kotlin still requires the `!!` lol. But I agree it should never happen, but if something gets broken and it does happen it's better to have the check right?

Reply:
Yeah, absolutely!